File: 2-medium/01441-build-an-array-with-stack-operations/solution.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        int i = 0;
        int num = 0;
        vector<string> res{};
        while (i < target.size()) {
            num += 1;
            res.push_back("Push");
            if (num == target[i]) {
                i += 1;
            } else {
                res.push_back("Pop");
            }
        }

        return res;
    }
};

File: 2-medium/01441-build-an-array-with-stack-operations/solution.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        i = 0
        num = 0
        res = []
        while i < len(target):
            num += 1
            res.append("Push")
            if num == target[i]:
                i += 1
            else:
                res.append("Pop")

        return res
